[DOCS-9959] Install the OP Worker (#28184)
DataDog/documentation: content/en/observability_pipelines/install_the_worker.md

---
title: Install the Worker
disable_toc: false
further_reading:
- link: "observability_pipelines/set_up_pipelines#set-up-a-pipeline"
  tag: "Documentation"
  text: "Set up pipelines"
---

## Overview

> The Observability Pipelines Worker is software that runs in your environment to centrally aggregate, process, and route your logs. You install and configure the Worker when you set up a pipeline. These are the general steps for setting up a pipeline in the UI:

Author: ```suggestion
The Observability Pipelines Worker is software that runs in your environment to centrally aggregate, process, and route your logs. You install and configure the Worker as part of the pipeline setup process. These are the general steps for setting up a pipeline in the UI:
```

These steps outline the setup process:

1. Select a log source.
1. Select destinations to which you want to send your logs.
1. Select and configure processors to transform your logs.
1. Install the Worker.
1. Deploy the pipeline.

**Note**: If you are using a proxy, see the `proxy` option in [Bootstrap options][1].

## Install the Worker

After you set up your source, destinations, and processors on the Build page of the pipeline UI, follow the steps on the Install page.

{{< img src="observability_pipelines/install_page.png" alt="The install page in the UI with a dropdown menu to choose your installation platform and fields to enter environment variables" style="width:100%;" >}}

1. Select the platform on which you want to install the Worker.
1. Enter the environment variables for your sources and destinations, if applicable.
1. Follow the instructions on installing the Worker for your platform. The command provided in the UI to install the Worker has the relevant environment variables populated.

{{< tabs >}}
{{% tab "Docker" %}}

{{% observability_pipelines/install_worker/docker %}}

{{% /tab %}}
{{% tab "Kubernetes" %}}

> {{% observability_pipelines/install_worker/kubernetes %}}

Author: Hey! The shortcode files already exist so I'm just referencing them.

> {{% observability_pipelines/install_worker/kubernetes %}}

Author: The shortcode files are already in the repo (they are being used in other pages as well), so I'm just referencing them, which is why you don't see the files themselves.

> {{% observability_pipelines/install_worker/kubernetes %}}

Author: I think this was the suggested edit: https://github.com/DataDog/documentation/pull/28184/commits/76c6c2c25513d2916d29b6b319b2e580ac4cdd13

    ```shell
    sudo apt-get update
    sudo apt-get install observability-pipelines-worker datadog-signing-keys
    ```
1. Add your keys, site (for example, `datadoghq.com` for US1), source, and destination environment variables to the Worker's environment file:
    ```shell
    sudo cat <<EOF > /etc/default/observability-pipelines-worker
    DD_API_KEY=<DATADOG_API_KEY>
    DD_OP_PIPELINE_ID=<PIPELINE_ID>
    DD_SITE=<DATADOG_SITE>
    <SOURCE_ENV_VARIABLES>
    <DESTINATION_ENV_VARIABLES>
    EOF
    ```
1. Start the worker:
    ```
    sudo systemctl restart observability-pipelines-worker
    ```

**Note**: The environment variables used by the Worker in `/etc/default/observability-pipelines-worker` are not updated on subsequent runs of the install script. If changes are needed, update the file manually and restart the Worker.

See [Update Existing Pipelines][1] if you want to make changes to your pipeline's configuration.

[1]: /observability_pipelines/update_existing_pipelines

{{% /tab %}}
{{% tab "RPM" %}}

If you prefer not to use the one-line installation script, follow these step-by-step instructions:

> 1. Set up the Datadog `rpm` repo on your system with the below command. **Note**: If you are running RHEL 8.1 or CentOS 8.1, use `repo_gpgcheck=0` instead of `repo_gpgcheck=1` in the configuration below.

Author: Fixed in this commit: https://github.com/DataDog/documentation/pull/28184/commits/76c6c2c25513d2916d29b6b319b2e580ac4cdd13
